### PR TITLE
Init ACRA and analytics onAttachBaseContext

### LINKED ISSUE #6882: ACRA should be initialized in attachBaseContext not onCreate
Per upstream docs (just noticed this!) ACRA should init in a different spot:

https://github.com/ACRA/acra/wiki/BasicSetup#configuring-acra---compile-time

We initialize it in onCreate, which is apparenty incorrect

https://github.com/ankidroid/Anki-Android/blob/3f5a293c9f3959f1d855cb3c7b3b3a9b6ad89fc7/AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.java#L233-L244

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description

According to ACRA documentation, we're meant to do the init earlier on.

https://github.com/ACRA/acra/wiki/BasicSetup#configuring-acra---compile-time

This should allow us to catch more exceptions in ContentProviders
Obtaining more Timber logs, and hopefully better crash reports

**Note: This will currently throw if the language init fails. I feel that the init method should not be allowed to fail**

## Fixes
Fixes #6882

## Approach
Remove the null `sInstance` check in `onCreate`, move the rest of the init to `onAttachBaseContext`

## How Has This Been Tested?

Unit Tested and my device still runs the app.

## Learning (optional, can help others)
https://github.com/ACRA/acra/wiki/BasicSetup#configuring-acra---compile-time

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by david-allison-1** [comment]
Tests pass on my physical Android, testing with emulator - glad we had them

**Comment by mikehardy** [comment]
> **Note: This will currently throw if the language init fails. I feel that the init method should not be allowed to fail**

Seems like the right stance in principle, I'm a fan of init never failing and relying on reasonable fallbacks, maybe with toasts or snackbars indicating there was something unexpected, if something unexpected happened, all while soldiering on.

Is there a reasonable fallback here? Is this what's blowing up every emulator?

**Comment by david-allison-1** [comment]
> > **Note: This will currently throw if the language init fails. I feel that the init method should not be allowed to fail**
>
> Is there a reasonable fallback here? Is this what's blowing up every emulator?

The method call is currently wrapped in a try..catch (Exception) returning a reasonable default, so will only fail on `Error`

**Comment by mikehardy** [comment]
Sounds good - then the only question is why the 100% androidTest meltdown :thinking:

**Comment by mikehardy** [comment]
@david-allison-1 seems like this one could be brought home if CI fail was investigated?

**Comment by david-allison-1** [comment]
> @david-allison-1 seems like this one could be brought home if CI fail was investigated?

I've had a second look and I'm still not at a point where I can repro this. 

I'd really like to get it done, but it's one that I don't have a solution for right now.

**Comment by david-allison-1** [comment]
Missed that we were doing a non-standard Application init process inside test setup. This should fix it.

**Comment by david-allison-1** [comment]
Turns out that there was two problems:

* `attachBaseContext` is not called on `setup()` each time for the tests, but `onCreate` is. We're reusing the `AnkiDroidApp` instance, so `attachBaseContext` can't be called twice
* There was a bug in our ACRA implementation - moving from "always" to "ask" would not re-enable the dialog.

**Comment by david-allison-1** [comment]
Additional request for feedback: I felt the refactorings in the ACRATest class were bordering on excessive.

**Review by mikehardy — APPROVED** [review]
excessive no way! you were following through where something seemed wrong and making the tests better, the refactors there aren't hard to follow and that's only the second time that object's been looked at (the ACRATest) so I'm not surprised the second time it saw attention, from a second pair of eyes there was a pile of low-hanging fruit.

Anyway, fantastic on the migration to preferred startup - which should result in better reporting - and squishing a bug that had unexpected/unwanted privacy control implications at the same time. This is behind the scenes but a great advance
